### PR TITLE
Created types for Openpay SDK functions

### PR DESCRIPTION
**Pull request by bjardon**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## openpay ([<kbd>unpkg</kbd>](https://unpkg.com/browse/openpay@latest/))
was missing the following properties:
1. BASE_URL
1. API_VERSION
1. SANDBOX_URL
1. SANDBOX_API_VERSION
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against caf25e7e7151bf83836720e6742e7d2675993a25
</p>

File: types/openpay/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (amcasey) [inline types/openpay/tsconfig.json:18]: Why was `esModuleInterop` required?

File: types/openpay/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,282 @@
+// Type definitions for openpay 1.0
+// Project: https://github.com/open-pay/openpay-node
+// Definitions by: Bruno Jardón <https://github.com/bjardon>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace openpay;
```
Comment (amcasey) [inline types/openpay/index.d.ts:6]: When using `export as namespace`, it's nice to add a corresponding test consuming the types that way.

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/openpay/tsconfig.json (hunk 1 above)
Comment (bjardon) [reply types/openpay/tsconfig.json:18]: The Typescript Docs say the `esModuleInterop` flag is required to import the package as `import Openpay from 'openpay'`. But I think it can also be imported using `import Openpay = require('openpay')`, right? Then I think the flag is not required. I was a little confused about this...

File: types/openpay/index.d.ts (hunk 2 above)
Comment (bjardon) [reply types/openpay/index.d.ts:6]: The module I'm writing types for is not a UMD module. I believe I left export by mistake.

**Comment by bjardon** [comment]
@amcasey I've addressed the comments you made on my code and will add 3 new commits with the corresponding fixes. Could you review them? Thx.

File: types/openpay/tsconfig.json (hunk 1 above)
Comment (amcasey) [reply types/openpay/tsconfig.json]: Yeah, it's super confusing.  Basically, you only want to use `esModuleInterop` in very specialized cases where the library works both with CommonJS-style imports (`const OpenPay = require`) and ES6-style imports (`import OpenPay from`), e.g. because it does fancy things with a bundler.  `import OpenPay = require` is a special TS-style import that "just works".

TL;DR: Please remove the flag if you're not aware of special characteristics of the library that make it necessary.

**Review by amcasey — APPROVED** [review]
Thanks for the changes!

File: types/openpay/tsconfig.json (hunk 1 above)
Comment (bjardon) [reply types/openpay/tsconfig.json]: Thank you for your clarification! Should I mark this conversation as resolved?

**Comment by bjardon** [comment]
Ready to merge
